evaluation/eval.py

```
@@ -176,15 +179,25 @@ async def ingredient_consistency(recipe: str, logger: logging.Logger) -> int:
     messages = format_message_history("ingredient_consistency", recipe=recipe)
 
     resp = await chatgpt.agenerate(messages=[messages])
-    log_messages(logger, "ingredient_consistency prompt:\n", messages)
-    logger.debug(f"ingredient_consistency response: {resp.generations[0][0].text}")
+    log_text = (
+        "ingredient_consistency prompt:\n"
+        + textwrap.indent("\n".join(msg.content for msg in messages), "  ")
+        + f"\ningredient_consistency response: {resp.generations[0][0].text}"
+    )
 
     response = resp.generations[0][0].text
     pattern = r"\d+"
     match = re.search(pattern, response)
 
-    out = int(match.group() if match else 0)
-    logger.debug(f"ingredient_consistency result: {out}")
+    try:
+        out = int(match.group())
+        logger.debug(f"{log_text}\n ingredient_consistency result: {out}\n\n\n")
+    except ValueError:
```

```suggestion
    if match is not None:
        out = int(match.group())
        logger.debug(f"{log_text}\n ingredient_consistency result: {out}\n\n\n")
    else:
```

---

Always better to avoid catching exceptions when you already know why the exception should happen. This way, exceptions that weren't what you expected can still surface, instead of disappearing silently